File: laboratories/ppo/exercises/lab10/java/Main.java
import java.lang.reflect.Method;
import java.util.Arrays;
import lab10.Actions.QuoteAction;
import lab10.Actions.VersionAction;
import lab10.Route;
import lab10.Router;

public class Main {
    public static void main(String[] args) throws Exception {
        Router router = new Router(new Route[]{
            new Route("version", VersionAction.class, "invoke"),
            new Route("quote", QuoteAction.class, "getRandomQuote"),
            new Route("quote:popular", QuoteAction.class, "getMostPopularQuote"),
            new Route("quote:id", QuoteAction.class, "getByIndex"),
        });

        Route route = router.match(args[0]);
        String[] parameters = Arrays.copyOfRange(args, 1, args.length);

        Object instance = route.getActionClass().getDeclaredConstructor().newInstance();
        Method method = route.getActionClass().getMethod(route.getMethod(), String[].class);
        String result = (String) method.invoke(instance, (Object) parameters);

        System.out.println(result);
    }
}

File: laboratories/ppo/exercises/lab10/python/main.py
import sys
from src.actions.QuoteAction import QuoteAction
from src.actions.VersionAction import VersionAction
from src.Route import Route
from src.Router import Router


router = Router([
    Route("version", VersionAction),
    Route("quote", QuoteAction, "get_random_quote"),
    Route("quote:popular", QuoteAction, "get_most_popular_quote"),
    Route("quote:id", QuoteAction, "get_by_index"),
])

route = router.match(sys.argv[1])

instance = route.action()
method = getattr(instance, route.method)
result = method(*sys.argv[2:])

print(result)
